inttest/common/footloosesuite.go

```diff
@@ -1289,18 +1289,34 @@ func (s *FootlooseSuite) getIPAddress(nodeName string) string {
 // CreateNetwork creates a docker network with the provided name, destroying
 // any network that has the same name first.
 func (s *FootlooseSuite) CreateNetwork(name string) error {
-	_ = s.DestroyNetwork(name)
+	// Don't create the network if it already exists as it might be in use from previous tests
+	// ran with K0S_KEEP_AFTER_TEST variable set.
+	if s.NetworkExists(name) {
+		return nil
+	}
 
 	cmd := exec.Command("docker", "network", "create", name)
 	return cmd.Run()
 }
 
 // DestroyNetwork removes a docker network with the provided name.
 func (s *FootlooseSuite) DestroyNetwork(name string) error {
+	// If we're supposed to leave the footloose containers running, don't try to destroy the network
+	if keepEnvironment(s.T()) {
```

Comment:
I'd have preferred to have this check made by the callers of this method. Seems kinda unexpected that the network won't be deleted by this function under certain circumstances. But yeah, I can currently see 6 call sites, so this is a pragmatic choice.

Ideally, the network lifecycle management would be done by the `FootlooseSuite` itself, i.e. it keeps track of the created networks and destroys them conditionally along with the rest of the cluster somewhere [here](https://github.com/k0sproject/k0s/blob/7ff1e012fd1cd02dab352e80bd3110691596133e/inttest/common/footloosesuite.go#L347-L356). This function would then become obsolete, as the inttests could simply rely on the fact that `FootlooseSuite` will take care of all the networks that it has created via `CreateNetwork`, instead of having to call `DestroyNetwork` manually in `TearDownSuite`.

Reply:
I don't like that a function called `DestroyNetwork` may be expected not to detroy the network, but changing this in every test makes it prone to forget to add this in some tests, not just now but also in the future.

A somewhat acceptable option which kind of solves both problems would be renaming the function to something like `MaybeDestroyNetwork` would solve the problem. It's not as good as handling this in `TearDownSuite`, but it's less work.

Reply:
Yeah, handling the conditional network removal on caller side would be bit painful. I renamed the func to `MaybeDestroyNetwork`

Reply:
Works for me. What do you think @twz123 ?

Reply:
Sure.